jetty-util/src/test/java/org/eclipse/jetty/util/TypeUtilTest.java

```diff
@@ -166,4 +172,14 @@ public void testGetLocation_JvmCore_Java8RT()
         String expectedJavaBase = "/rt.jar";
         assertThat(TypeUtil.getLocationOfClass(String.class).toASCIIString(),containsString(expectedJavaBase));
     }
+
+    @Test
+    @EnabledOnJre(JRE.JAVA_8)
+    public void testGetLocation_JavaLangThreadDeath_Java8RT()
+    {
+        // Class from JVM core
+        String expectedJavaBase = "/rt.jar";
+        assertThat(TypeUtil.getLocationOfClass(java.lang.ThreadDeath.class).toASCIIString(),containsString(expectedJavaBase));
+    }
+    */
```

Comment:
These will be removed, I just needed them temporarily around for discussing the heart of the issue and a possible fix.

Reply:
While it is not vital.... it would be really good if when running on java8 we could still work out that `java.lang.String` comes from `rt.jar`.   We do have other sources of boot classes (eg our old NPN and ALPN implementations), so it can be meaningful to differential.    So surely there is some way to do this in the cases that recursion will not be a problem.... but if not, then null is fine until we are all on java >8

Reply:
I'll see if a happy solution can be found.

Reply:
New solution available in commit 8d59302978d8107bfce40754ead29986940b5f94